diff --git a/node_modules/preact/compat/server.js b/node_modules/preact/compat/server.js
index 3eaf720..3231a4e 100644
--- a/node_modules/preact/compat/server.js
+++ b/node_modules/preact/compat/server.js
@@ -19,8 +19,19 @@ try {
 	);
 }
 
+var renderToReadableStream;
+try {
+	const mod = require('preact-render-to-string/stream');
+	renderToReadableStream = mod.default || mod.renderToReadableStream || mod;
+} catch (e) {
+	throw Error(
+		'renderToReadableStream() error: update "preact-render-to-string" dependency to at least 6.5.0.'
+	);
+}
+
 module.exports = {
 	renderToString: renderToString,
 	renderToStaticMarkup: renderToString,
-	renderToPipeableStream: renderToPipeableStream
+	renderToPipeableStream: renderToPipeableStream,
+	renderToReadableStream
 };
diff --git a/node_modules/preact/package.json b/node_modules/preact/package.json
index f92ca82..324d03a 100644
--- a/node_modules/preact/package.json
+++ b/node_modules/preact/package.json
@@ -84,6 +84,11 @@
       "import": "./compat/server.mjs",
       "require": "./compat/server.js"
     },
+    "./compat/server.browser": {
+      "browser": "./compat/server.browser.js",
+      "import": "./compat/server.mjs",
+      "require": "./compat/server.js"
+    },
     "./compat/jsx-runtime": {
       "types": "./jsx-runtime/src/index.d.ts",
       "import": "./compat/jsx-runtime.mjs",
