diff --git a/node_modules/preact/compat/server.js b/node_modules/preact/compat/server.js
index 7156872..a222ac6 100644
--- a/node_modules/preact/compat/server.js
+++ b/node_modules/preact/compat/server.js
@@ -9,7 +9,21 @@ try {
 	);
 }
 
+function renderToReadableStream(element) {
+	return Promise.resolve().then(function() {
+		const stream = new ReadableStream({
+			start(controller) {
+				controller.enqueue(new TextEncoder().encode(renderToString(element)));
+				controller.close();
+			}
+		});
+		stream.allReady = true;
+		return stream;
+	});
+}
+
 module.exports = {
 	renderToString: renderToString,
+	renderToReadableStream: renderToReadableStream,
 	renderToStaticMarkup: renderToString
 };
diff --git a/node_modules/preact/package.json b/node_modules/preact/package.json
index 688d96e..5f14a14 100644
--- a/node_modules/preact/package.json
+++ b/node_modules/preact/package.json
@@ -75,6 +75,11 @@
       "import": "./compat/server.mjs",
       "require": "./compat/server.js"
     },
+    "./compat/server.browser": {
+      "browser": "./compat/server.browser.js",
+      "import": "./compat/server.mjs",
+      "require": "./compat/server.js"
+    },
     "./compat/jsx-runtime": {
       "types": "./jsx-runtime/src/index.d.ts",
       "import": "./compat/jsx-runtime.mjs",
